auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java

```diff
@@ -269,32 +269,110 @@ public String getRefreshToken() {
 
 
     /**
-     * Returns the custom attribute.
+     * Returns the Identity JWT.
      *
-     * @param attribute the attribute name to retrieve
-     * @return the custom attribute.
+     * @return the identity JWT.
      */
-    @Override
-    public String getCustomAttribute(final String attribute) {
-        String attributeValue = null;
+    private JSONObject getRawIdentityToken() {
+        JSONObject identityJWT = null;
         try {
             String identityToken = getIdentityToken();
             JsonWebSignature signature = new JsonWebSignature();
             signature.setCompactSerialization(identityToken);
             String decoded = signature.getUnverifiedPayload();
             try {
-                JSONObject rawIdentityToken = new JSONObject(decoded);
-                if (rawIdentityToken.has(attribute)) {
-                    attributeValue = rawIdentityToken.getString(attribute);
-                } else {
-                    Log.d("Invalid Path", "No Object Exists in the JSON Path " + attribute);
-                }
+                identityJWT = new JSONObject(decoded);
             } catch (JSONException e) {
                 Log.d("Error Getting Attribute", e.getMessage());
             }
         } catch (JoseException e) {
             Log.d("Error Getting Attribute", e.getMessage());
         }
-        return attributeValue;
+        return identityJWT;
+    }
+
+    private boolean customAttributeExists(String attributeName) {
+        JSONObject jwt = getRawIdentityToken();
+        return jwt != null && jwt.has(attributeName);
+    }
+
+    /**
+     * Returns the custom user attribute of type string.
+     *
+     * @param attributeName the user attribute to retrieve
+     * @return the custom string attribute
+     */
+    public String getCustomStringAttribute(String attributeName) {
+        String attribute = null;
+        JSONObject jwt = getRawIdentityToken();
+
+        if (customAttributeExists(attributeName)) {
+            try {
+                attribute = jwt.getString(attributeName);
```

Comment:
you can use `optString/optBoolean/optLong/optInt` methods. This way you don't need to catch the exceptions.

Reply:
Ah, nice. Updated in https://github.com/aerogear/aerogear-android-sdk/pull/149/commits/2705672cc12346f6e67904120e68123420d069e1